embedded-graphics/src/draw_target.rs

```diff
@@ -222,161 +249,78 @@ where
     /// [`core::convert::Infallible`]: https://doc.rust-lang.org/stable/core/convert/enum.Infallible.html
     type Error;
 
-    /// Draws a pixel on the display.
-    fn draw_pixel(&mut self, item: drawable::Pixel<C>) -> Result<(), Self::Error>;
-
-    /// Draws an object from an iterator over its pixels.
-    fn draw_iter<T>(&mut self, item: T) -> Result<(), Self::Error>
-    where
-        T: IntoIterator<Item = drawable::Pixel<C>>,
-    {
-        for pixel in item {
-            self.draw_pixel(pixel)?;
-        }
-
-        Ok(())
-    }
-
     /// Returns the dimensions of the `DrawTarget` in pixels.
-    fn size(&self) -> Size;
-
-    /// Clears the display with the supplied color.
     ///
-    /// This default implementation can be replaced if the implementing driver provides an
-    /// accelerated clearing method.
-    fn clear(&mut self, color: C) -> Result<(), Self::Error>
-    where
-        Self: Sized,
-    {
-        primitives::Rectangle::new(Point::zero(), self.size())
-            .into_styled(PrimitiveStyle::with_fill(color))
-            .draw(self)
-    }
+    /// This should return the size of the visible area of the display, starting at `(0, 0)`. If a
+    /// display supports drawing to pixels outside this area, that area should not be reported in
+    /// calls to `size`.
```

Comment:
I think this needs some clarification.

Does this mean that it should also only return the visible area if I use it in conjunction with a larger framebuffer in the controller memory to support scrolling?

Reply:
Exactly that, yes. What would you add  to the comment to explain this better?

Reply:
I'm not sure. There might be situations where a target is larger than the actual display and where it would make sense for `size` to return a size larger than the display. If, for example, a larger framebuffer (independend of the one used by the display/driver) is used to support scrolling an UI layout algorithm might need to know the available space in that buffer.

Reply:
The main purpose I have in mind for the return value of this method is for doing layout in the visible area of the screen, but we do need to be able to get the renderable area for proper clipping and scrolling as you mention. How about having two methods, `visible_size` and `render_size` for visible and total pixel area respectively where `render_size`' default impl simply calls visible size?

Reply:
I'm not sure this solves the problem, if `visible_size` is `< render_size` the layout algorithm still wouldn't know where the top left coordinate of the visible window is.

Reply:
If the visible area top left is `(0, 0)`, the off-screen area could be a `Rectangle` with a negative `top_left` value. Would that work?. This way, it keeps the drawable area positioned within the render area at a known offset whilst keeping the origin at `(0, 0)` which is what I think most end users will want. 

Also, AFAIK it's more common to have a display with `visible_size == render_size` rather than `visible_size < render_size`, so to me it makes sense to use `(0, 0)` as the _visible_ origin.

Reply:
> If the visible area top left is `(0, 0)`, the off-screen area could be a `Rectangle` with a negative `top_left` value. Would that work?. This way, it keeps the drawable area positioned within the render area at a known offset whilst keeping the origin at `(0, 0)` which is what I think most end users will want.

Yes, this is what most people want A better way to implement this might be to add a `TranslatedDrawTarget` wrapper (which I suggested some time ago, but I have no idea where). The display driver would implement a basic draw target which covers the entire framebuffer and reports it's size in the `size` function. This draw target can then be wrapped, which causes all draw operations to work as expected by using coordinates starting from `(0, 0)`.

Reply:
That sounds like a good idea. For this PR, I think the best approach is to change the `size` docs to mention it should return the entire drawable area and add `TranslatedDrawTarget` in a later PR. Shall I open a ticket to add it once this PR is merged?

Reply:
Yes, let's try this approach and see if it works out in practice.

Reply:
Ok, sounds good. #355 opened to track. I'll change the docs in this PR momentarily.